File: src/Client/Logistics.OfficeApp/src/app/core/enums/paymentFor.ts
import {EnumLike} from './enumLike';

export enum PaymentFor {
  Payroll,
  Subscription,
  Invoice,
  Other
}

export const PaymentForEnum: EnumLike = {
  Payroll: {value: 0, description: 'Payroll'},
  Subscription: {value: 1, description: 'Subscription'},
  Invoice: {value: 2, description: 'Invoice'},
  Other: {value: 3, description: 'Other'}
};

File: src/Shared/Logistics.Shared.Consts/Enums/PaymentFor.cs
using System.ComponentModel;
using System.Runtime.Serialization;

namespace Logistics.Shared.Consts;

public enum PaymentFor
{
    [Description("Employee Payroll"), EnumMember(Value = "employee_payroll")]
    Payroll,
    
    [Description("Subscription"), EnumMember(Value = "subscription")]
    Subscription,
    
    [Description("Invoice"), EnumMember(Value = "invoice")]
    Invoice,
    
    [Description("Other"), EnumMember(Value = "other")]
    Other
}
